### PR TITLE
feat: proxy headers

### LINKED ISSUE #130: Add header on proxy requests
I propose to add a 'headers' property on proxy section to define headers to be injected in the request before sending to proxy destination.

For example:
```
proxy:
    host: https://example.com
    headers:
        X-Forwared-Proto: http
```

If ok, I can work on a pull request.

### PR DESCRIPTION
**Pull request by pplr**
Fix #130

### CODE REVIEW

File: server/types/mock.go [hunk 1]
```diff
@@ -163,10 +164,15 @@ func (mp MockProxy) Redirect(req Request) (*MockResponse, error) {
...
 		return nil, err
 	}
-	proxyReq.Header = req.Headers
+	proxyReq.Header = req.Headers.Clone()
 	if mp.KeepHost {
 		proxyReq.Host = req.Headers.Get("Host")
 	}
+	for key, values := range mp.Headers {
+		for _, value := range values {
+			proxyReq.Header.Add(key, value)
```
Comment (gwleclerc) [inline server/types/mock.go:173]: By doing that if a header is already setted in request, the value of proxy header will be added and will not replace it.
And I don't think it's the desired behavior

Comment (gwleclerc) [reply server/types/mock.go:173]:
```go
proxyReq.Header[key] = values
```

Comment (pplr) [reply server/types/mock.go:174]: I added `proxyReq.Header.del(key)` in the keys loop to clear any present header. `proxyReq.Header[key] = values` bypasses header key canonicalization. As a consequence, header may not be overwritten if key is specified with a different case in proxy definition and in incoming request.

File: docs/technical-documentation/mock-definition.md [hunk 2]
```diff
@@ -313,10 +313,15 @@ proxy:
...
 By default, redirect responses from the destination host are returned as any other response. Setting `follow_redirect` to `true`
 makes Smocker follow any redirect response before responding.
 
 Host header is overriden using destination host value by default. With `keep_host` set to `true`, request sent to the 
-destination host have same `Host` HTTP header as incoming request.
\ No newline at end of file
+destination host have same `Host` HTTP header as incoming request.
+
+Headers defined in the proxy mock definition are injected in the request sent to the destination host. If the header is already
+present in the incoming request, its value is overwritten.
```
Comment (gwleclerc) [inline docs/technical-documentation/mock-definition.md:327]:
```suggestion
present in the incoming request, its value will be overwritten.
```

**Comment by gwleclerc** [comment]
LGTM

**Pull request merged**